utils.sh

```diff
@@ -266,11 +266,13 @@ function sync_repo_and_patch {
 
     pushd $DEST
 
+    DEFAULT_BRANCH=$(git remote show origin | sed -n '/HEAD branch/s/.*: //p')
```

Comment:
What about using `git ls-remote` to check if the remote branch exists or not, ie `git ls-remote -q --exit-code --heads $1 main`?

Reply:
But then we'd be back to hard-coding a branch name, vs this approach which finds the HEAD branch from the origin remote (which should handle any branch name)?

Reply:
Agreed. The only benefit of the ls-remote is that it doesn't require to clone the repo for the check, but in this case it's defintely not required.